File: SOL.Identity/SOL.Identity.Application/User/CreateUserCommandHandler.cs
using System.Threading;
using System.Threading.Tasks;
using FluentValidation;
using MediatR;
using SOL.Core.Commands;
using SOL.Core.Interfaces;
using SOL.Identity.SOL.Identity.Domain.Commands.User;
using SOL.Identity.SOL.Identity.Domain.Interfaces.Repositories;

namespace SOL.Identity.SOL.Identity.Application.User
{
    public class CreateUserCommandHandler : CommandHandlerBase, ICommandHandlerAsync<CreateUserCommand>, IRequestHandler<CreateUserCommand, Result>
    {
        private readonly IValidator<CreateUserCommand> _createUserCommandValidator;
        private readonly IUserRepository _userRepository;

        public CreateUserCommandHandler(IValidator<CreateUserCommand> validator, IUserRepository userRepository)
        {
            _createUserCommandValidator = validator;
            _userRepository = userRepository;
        }

        public async Task<Result> Handle(CreateUserCommand command, CancellationToken cancellationToken)
        {
            var validationResult = Validate(command, _createUserCommandValidator);

            if (validationResult.IsValid)
            {
                Domain.Entities.User user = new() { Email = command.Email, UserName = command.Email, Password = command.Password };
                await _userRepository.AddAsync(user);
            }

            return Return();
        }
    }
}

File: SOL.Web/src/app/application/User/CreateUserCommandHandler.ts
import { injectable, inject } from "tsyringe";
import { validate, IsString, Length } from "class-validator";
import ICommandHandlerBase from "@/app/core/commands/ICommandHandlerBase";
import IUserRepository from "@/app/domain/interfaces/repositories/IUserRepository";
import IUser from "@/app/domain/entities/User";
import Result from "@/app/core/commands/Result";
import CreateUserCommand from "@/app/domain/commands/User/CreateUserCommand";
import createUserCommandValidator from "@/app/domain/commands/User/Validators/CreateUserCommandValidator";
/*
 * Command handler resolution:
 * CAS: 1 command 1 action 1 side
 * Add and push can be one 1 command
 * Update and push update can be one 1 command
 * ... other commands are similar
 * Add/Delete/Get/Update work with store.
 * Push/RequestDelete/Fetch/RequestUpdate work with server.
 * Command always return Result.
 * */
@injectable()
export default class CreateUserCommandHandler implements ICommandHandlerBase {
  constructor(
    @inject("UserRepository") private userRepository: IUserRepository // Late @inject("UserRepository") // private createUserCommandValidator: IUserRepository
  ) {}

  async handle(command: CreateUserCommand) {
    // Validation.
    const validationResult = await createUserCommandValidator.Validate(command);
    if (validationResult.length > 0) {
      return new Result(false, validationResult);
    }
    // Create entity.
    const user: IUser = {
      Id: "something id",
      Email: command.Email,
      Password: command.Password,
    };
    // If you add entity to store use Entity argument.
    await this.userRepository.AddAsync(user);
    // Actions with server should be without arguments.
    await this.userRepository.PushUserAsync();
    // Return Result().
    return new Result(true, validationResult);
  }
}
